vault/logical_system.go

```diff
@@ -1717,6 +1733,118 @@ func (b *SystemBackend) handleTuneWriteCommon(ctx context.Context, path string,
 		}
 	}
 
+	// user-lockout config
+	{
+		var apiuserLockoutConfig APIUserLockoutConfig
+
+		userLockoutConfigMap := data.Get("user_lockout_config").(map[string]interface{})
+		var err error
+		// Augmenting userLockoutConfigMap for some config options to treat them as comma separated entries
+		if userLockoutConfigMap != nil && len(userLockoutConfigMap) != 0 {
+			err := mapstructure.Decode(userLockoutConfigMap, &apiuserLockoutConfig)
+			if err != nil {
+				return logical.ErrorResponse(
+						"unable to convert given user lockout config information"),
+					logical.ErrInvalidRequest
+			}
+
+			// Supported auth methods for user lockout configuration: ldap, approle, userpass
+			// "all" is used to apply the configuration to all supported auth methods
+			switch strings.ToLower(mountEntry.Type) {
+			case "all", "ldap", "approle", "userpass":
+			default:
+				return logical.ErrorResponse("tuning of user lockout configuration for auth type %q not allowed", mountEntry.Type),
+					logical.ErrInvalidRequest
+
+			}
+		}
+
+		if len(userLockoutConfigMap) > 0 && mountEntry.Config.UserLockoutConfig == nil {
+			mountEntry.Config.UserLockoutConfig = &UserLockoutConfig{}
+		}
+
+		var oldUserLockoutThreshold int64
+		var newUserLockoutDuration, oldUserLockoutDuration time.Duration
+		var newUserLockoutCounterReset, oldUserLockoutCounterReset time.Duration
+		var oldUserLockoutDisable bool
+
+		if _, ok := userLockoutConfigMap["lockout_threshold"]; ok {
```

Comment:
nit: It seems a bit strange that we are checking on `userLockoutConfigMap`, but then using `apiuserLockoutConfig`. Why not use the latter to check if the specific entry is not nil or something?

Reply:
It was because I had disable lockout as a bool value. Therefore, checking apiuserLockoutConfig will not work as expected. I modified the apiuserLockoutConfig struct to use *bool to check for nil values. I will change checking userLockoutConfigMap to checking apiuserLockoutConfig